src/client/GameWindow.cpp

```diff
@@ -52,72 +53,73 @@ namespace client
             {
-                clientGameWindow->draw(*priorityCardTexture.body);
+                gameEnginePtr->clientWindow->draw(*priorityCardTexture.body);
             }
         }
 
-        boxTexture->drawTextureDisplayerSprite(clientGameWindow);
+        boxTexture->drawTextureDisplayerSprite(gameEnginePtr->clientWindow);
 
         for (unsigned i = 0; i < actionCardsToDisplay.size(); i++)
         {
-            clientGameWindow->draw(actionCardsToDisplay[i].texture->getSprite(0));
-            clientGameWindow->draw(*actionCardsToDisplay[i].title);
-            clientGameWindow->draw(*actionCardsToDisplay[i].body);
+            gameEnginePtr->clientWindow->draw(actionCardsToDisplay[i].texture->getSprite(0));
+            gameEnginePtr->clientWindow->draw(*actionCardsToDisplay[i].title);
+            gameEnginePtr->clientWindow->draw(*actionCardsToDisplay[i].body);
         }
 
         for (unsigned i = 0; i < whoIsPlayingButtons.size(); i++)
         {
-            clientGameWindow->draw(*whoIsPlayingButtons[i].buttonRect);
-            clientGameWindow->draw(*whoIsPlayingButtons[i].buttonText);
+            gameEnginePtr->clientWindow->draw(*whoIsPlayingButtons[i].buttonRect);
+            gameEnginePtr->clientWindow->draw(*whoIsPlayingButtons[i].buttonText);
         }
 
-        clientGameWindow->draw(hudTextureToDisplay.at(TURN_NUMBER % 5).getSprite());
+        gameEnginePtr->clientWindow->draw(hudTextureToDisplay.at(TURN_NUMBER % 5).getSprite());
 
         for (unsigned i = 5; i < hudTextureToDisplay.size(); i++)
         {
-            hudTextureToDisplay[i].drawTextureDisplayerSprite(clientGameWindow);
+            hudTextureToDisplay[i].drawTextureDisplayerSprite(gameEnginePtr->clientWindow);
         }
 
-        clientGameWindow->display();
+        gameEnginePtr->clientWindow->display();
     }
 
     /*!
-     * @brief Loop that look for events to happend and call displayWindow()
-     * @param clientWindow is window that comes from the engine
-     * @param quitGame is the function used to quit the menu, it is load as an attribut
-     * @param callback is the function used to return where the user click on the screen
-     */
-    void GameWindow::startGame(std::shared_ptr<sf::RenderWindow> clientWindow, std::function<void(bool)> quitGame, std::function<void(int, int)> callback)
+    * @brief Loop that look for events to happend and call displayWindow()
+    * @param clientWindow is window that comes from the engine
+    * @param quitGame is the function used to quit the menu, it is load as an attribut
+    * @param callback is the function used to return where the user click on the screen
+    */
+    void GameWindow::startGame()
     {
-        quitGameWindow = quitGame;
-        clickEvent = callback;
-        clientGameWindow = clientWindow;
+        if (gameEnginePtr == nullptr)
+        {
+            return;
+        }
 
         std::shared_ptr<bool> moveMode = std::make_unique<bool>(false);
```

Comment:
I know it was already there but there is an issue here:
`std::shared_ptr<bool> moveMode = std::make_shared<bool>(false);`
or
`std::unique_ptr<bool> moveMode = std::make_unique<bool>(false);` but not a mix of both

Reply:
Done